### DISCUSSION TITLE
NFT buy back mechanism from the SFL treasury

### DISCUSSION THREAD

**Discussion by Bold-maverick** [sunflower-land/sunflower-land#1282] [Ideas]
Hello fellow farmers,

I was wondering If you're considering implementing a buy-back mechanism for NFTs?

For example, if a player buys an NFT on secondary and then wants SFL, or resources instead, then the treasury can buy it back.

I think this will provide more arbitrage opportunities, reduces the supply of some NFTs, and make trading more fun.

let me know what you think.

**Comment by Yougottabekiddingme123** [comment]
I think this is a good suggestion and a much better use case instead of the giveaways on twitch and discord. The Goblin fee and listing fee should be used to buy back nfts. Many nfts have dropped in value and are now cheaper than the resources needed to produce them.

**Comment by khunso-id10440** [comment]
thats sound good !

**Comment by adamhannigan** [comment]
We are open to any possibilities. The treasury should be used for whatever improves the game and the community.

Do you have some specific examples of how this would work. What do you think the positives and negatives of this feature could be?

**Comment by BerJack43** [comment]
I think it's a great idea. Because the finished harvest will lose its value with each halving. So the Goblin Tressure can convert this effect into SFL for the benefit of the community. 

However, I see a big problem, the disputes therein:

If you sell the Croops before halving, for example, through a mechanism, you get SFLs. Throwing these SFLs back into the trade carries great risks and possibilities that the SFL value would decrease. However, one assumes that a Mint must take place before, so that one can sell also evenly the necessary NFTs to the treasury. Well, a fee with 20% should be taken; Means in reverse: If I acquire a NFT with 100 gold that has a value of, for example, 100$; So I must first afford the 100 gold, so the 100 SFL or earned money to get me this NFT. If I have no need for it and want to sell it, I can sell it for 80 gold, for example. 
By buying it back with the same resources would ensure that the value of the SFL remains the same and leads to no dilution.

The resulting advantages would be:

Example: 

1st NFT bought for 100 gold, sold for 80:
- One can acquire this NFT again
- By acquiring it, the new buyer has to pay 100 gold again

Additionally one could add: That the value of the NFTs increases every month by 1%, so that more than the original cost is paid so instead of 100 equal to 101 in the following month or so.

Whenever something sold back is bought again: the resulting profits, in this case 20 gold, go into the goblin's coffers.

That means: If something is sold, the necessary material can be provided from the goblin cash; If the NFT bought from the goblin cash is sold again, the resulting profit goes directly into the cash. In this way, the value of the goblin fund can circulate.
